File: from-2501-to-2600/2558-take-gifts-from-the-richest-pile/solution.go
func pickGifts(gifts []int, k int) int64 {
    var result int64 = 0
    for i := 0; i < len(gifts); i ++ {
        result += int64(gifts[i])
    }
    
    for t := 0; t < k; t ++ {
        maxIdx := -1
        maxGft := -1
        for i := 0; i < len(gifts); i ++ {
            if gifts[i] > maxGft {
                maxGft = gifts[i]
                maxIdx = i
            }
        }
        sqrt := int(math.Floor(math.Sqrt(float64(gifts[maxIdx]))))
        result -= int64(gifts[maxIdx] - sqrt)
        gifts[maxIdx] = sqrt
    }
    return result
}

File: from-2501-to-2600/2558-take-gifts-from-the-richest-pile/solution.py
class Solution:
    def pickGifts(self, gifts: List[int], k: int) -> int:
        gifts = [-g for g in gifts]
        heapq.heapify(gifts)

        for _ in range(k):
            g = -1 * math.floor(math.sqrt(-1 * heapq.heappop(gifts)))
            heapq.heappush(gifts, g)

        return -sum(gifts)